lib/arrays.md

```diff
@@ -5,44 +5,59 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
+An array is a collection of ordered objects.  They are indexed to an integer, and are thus stuck in place (unless the array is edited).  So, the first object in the array is indexed to 0, the second object is indexed to 1, and so on.  The end of the array can be counted backwards from -1.  Arrays can be multidimensional (having any number of n-tuples) assigned to the integers.  Like [4,23,27] could be a single object in the array (properly formatted, of course).
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+As stated above, multidimensional objects are suited for arrays.  Another type of collection that would do well in arrays would be lists that the user needs to have in a specific order (for instance, the list of persons in the order that they arrived so that they might be able to give a door prize to the first persons that had arrived).  
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+Let's say that we have an array with the name alphabet, with the following collection of objects: alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm'].  To retrieve the sixth's element of that array, you use: alphabet[6], which will return g in this case.  If I tried to input alphabet[9999], I would get a nil return, or the default value.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+It is possible to find the -6th element in the array.  Arrays can identify the element by counting backwards.  The negative means that the count will start from the end of the array.  In the case of alphabet, it will return h as value. 
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+A way to perform an operation on every element inside an array would be to use the map! function. Map! allows for the new edited versions of the elements to replace the elements in their place.  So, if we were to write alphabet.map! { |x| x + "!" } it would return #=> ['a!', 'b!', 'c!', 'd!', 'e!', 'f!', 'g!', 'h!', 'i!', 'j!', 'k!', 'l!', 'm!'].
```

Comment:
Does `map!` permanently alter the original array? Are there times that we might _not_ want to do this? In those times, what could we use instead of `map!`?